realm/realm-annotations-processor/src/main/java/io/realm/processor/ClassMetaData.java

```diff
@@ -441,13 +459,13 @@ private boolean categorizeIndexField(Element element, VariableElement variableEl
     private void categorizeRequiredField(Element element, VariableElement variableElement) {
         if (Utils.isPrimitiveType(variableElement)) {
             Utils.error(String.format(Locale.US,
-                    "@Required annotation is unnecessary for primitive field \"%s\".", element));
+                    "@Required and @NotNull annotation is unnecessary for primitive field \"%s\".", element));
             return;
         }
 
-        if (Utils.isRealmList(variableElement) || Utils.isRealmModel(variableElement)) {
+        if (Utils.isRealmModel(variableElement)) {
```

Comment:
why removing the `isRealmList ` check?

Reply:
Because it conflicted with Kotlin mostly. RealmList is always non-null for managed object. I don't know why we didn't allow this to be set. At worst it would be redundant information.